packages/devtools_app/web/index.html

```diff
@@ -45,88 +45,36 @@
-        // unregister any service workers:
-        loadMainDartJs();
-        navigator.serviceWorker.getRegistrations().then(function (registrations) {
-          for (let registration of registrations) {
-            registration.unregister();
-          }
-        });
-      } else {
-        // Service workers are supported. Use them.
-        window.addEventListener('load', function () {
-          // Wait for registration to finish before dropping the <script> tag.
-          // Otherwise, the browser will load the script multiple times,
-          // potentially different versions.
-          var serviceWorkerUrl = 'service_worker.js?v=' + version;
-          navigator.serviceWorker.register(serviceWorkerUrl)
-            .then((reg) => {
-              function waitForActivation(serviceWorker) {
-                serviceWorker.addEventListener('statechange', () => {
-                  if (serviceWorker.state == 'activated') {
-                    loadMainDartJs();
-                  }
-                });
-              }
-              if (!reg.active && (reg.installing || reg.waiting)) {
-                // No active web worker and we have installed or are installing
-                // one for the first time. Simply wait for it to activate.
-                waitForActivation(reg.installing || reg.waiting);
-              } else if (!reg.active.scriptURL.endsWith(version)) {
-                // When the app updates the version changes, so we
-                // need to ask the service worker to update.
-                reg.update();
-                waitForActivation(reg.installing);
-              } else {
-                // Existing service worker is still good.
-                loadMainDartJs();
-              }
-            })
-            .catch((err) => {
-              console.warn(
-                ` Falling back to plain <script> tag. Error loading service worker: ${err}`);
-              loadMainDartJs();
-            });
+</head>
 
-          // If service worker doesn't succeed in a reasonable amount of time,
-          // fallback to plain <script> tag.
-          setTimeout(() => {
-            if (!scriptLoaded) {
-              console.warn(
-                'Failed to load app from service worker. Falling back to plain <script> tag.',
-              );
-              loadMainDartJs();
-            }
-          }, 4000);
-        });
-      }
-    } else {
-      // Service workers not supported. Just drop the <script> tag.
-      loadMainDartJs();
-    }
+<body>
+
+  <script>
+    window.addEventListener('load', function(ev) {
+      // Download main.dart.js
+      _flutter.loader.loadEntrypoint({
+        serviceWorker: {
+          serviceWorkerVersion: serviceWorkerVersion,
+        },
+        onEntrypointLoaded: function(engineInitializer) {
+          engineInitializer.initializeEngine({
+            renderer: 'canvaskit', 
+            canvasKitBaseUrl: 'canvaskit'
```

Comment:
curious, does the call to initializeEngine work without setting the base url? In the example on the flutter website, I only saw "renderer" specified, so I'm wondering why we need this too

Reply:
Yep, looks like this is required otherwise it tries to download canvaskit from the CDN, which is surprising. Opened https://github.com/flutter/flutter/issues/121743 in Flutter